2_process.R

```diff
@@ -62,6 +69,26 @@ p2_targets_list <- list(
     p2_site_list_nontidal_csv,
     create_site_list_nontidal(p2_wqp_SC_filtered,p1_nwis_sites,p1_daily_data,p1_inst_data,
                               hucs=drb_huc8s,crs_out="NAD83",p2_sites_w_segs,"2_process/out/DRB_SC_sitelist_nontidal.csv"),
-    format = "file")
-
+    format = "file"),
+  
+  # # target for NADP 
+  tar_target(p2_NADP_Data,
+             lapply(list.files(path = p1_NADP_data_unzipped, full.names = T), function(x) read.csv(x, sep = ',') %>%
```

Comment:
Do you think this function could be useful for processing other NHD attributes? It seems like it could be for any that have multiple years. It could be worth pulling it into a named function instead of having it within this target

Reply:
Okay, sounds good. No need to make a function for the lapply portion because that's already about as concise as one could have. And the filter is also concise enough to add whenever it's needed, so I'd say also no need for a function there.

> Another idea would be to create a function that subsets by comids in given aoi and runs the xwalk to PRMS then, because all NHD datasets require xwalk from comid to PRMS segid

That would be good for the next PR that aggregates from the NHD to the PRMS data. The subset to AOI is good to keep in this PR, though.

Reply:
Ok 👍